### ISSUE TITLE
ACL is applied when method is POST, but not GET

### ISSUE THREAD

**Issue by csvan** [strongloop/loopback#1686]
I have the following relevant ACLs on one of my models:

```
    {
      "accessType": "*",
      "principalType": "ROLE",
      "principalId": "$everyone",
      "permission": "DENY"
    },
    {
      "accessType": "EXECUTE",
      "principalType": "ROLE",
      "principalId": "$everyone",
      "permission": "ALLOW",
      "property": "sendVerificationEmail"
    }
```

Now, this will work fine if I define `sendVerificationEmail` like so:

```
  MyModel.remoteMethod('sendVerificationEmail',
    {
      http: {path: '/sendVerificationEmail', verb: 'post'},
      accepts: [
        {
          arg: 'id',
          type: 'string',
          required: true,
          description: 'The id of the user.'
        }
      ],
      returns: {
        arg: 'status',
        type: 'json'
      }
    });
```

In this case, the ACL log is the following for an invocation:

```
  loopback:security:acl The following ACLs were searched:  +0ms
  loopback:security:acl ---ACL--- +1ms
  loopback:security:acl model MyModel +0ms
  loopback:security:acl property sendVerificationEmail +0ms
  loopback:security:acl principalType ROLE +0ms
  loopback:security:acl principalId $everyone +0ms
  loopback:security:acl accessType EXECUTE +0ms
  loopback:security:acl permission ALLOW +0ms
  loopback:security:acl with score: +0ms 8132
  loopback:security:acl ---ACL--- +1ms
  loopback:security:acl model MyModel +1ms
  loopback:security:acl property * +0ms
  loopback:security:acl principalType ROLE +0ms
  loopback:security:acl principalId $everyone +0ms
  loopback:security:acl accessType * +0ms
  loopback:security:acl permission DENY +0ms
  loopback:security:acl with score: +0ms 7495
  loopback:security:acl ---ACL--- +1ms
  loopback:security:acl model MyModel +0ms
  loopback:security:acl property * +0ms
  loopback:security:acl principalType ROLE +0ms
  loopback:security:acl principalId $everyone +0ms
  loopback:security:acl accessType * +0ms
  loopback:security:acl permission DENY +0ms
  loopback:security:acl with score: +0ms 7495
  loopback:security:acl ---Resolved--- +0ms
```

However, I get 401:s if I do the following:

```
  MyModel.remoteMethod('sendVerificationEmail',
    {
      http: {path: '/:id/sendVerificationEmail', verb: 'get'},
      accepts: [
        {
          arg: 'id',
          type: 'string',
          required: true,
          http: {source: 'path'},
          description: 'The id of the user.'
        }
      ],
      returns: {
        arg: 'status',
        type: 'json'
      }
    });
```

In this case the ACL log is:

```
  loopback:security:acl The following ACLs were searched:  +0ms
  loopback:security:acl ---ACL--- +1ms
  loopback:security:acl model MyModel +1ms
  loopback:security:acl property * +0ms
  loopback:security:acl principalType ROLE +0ms
  loopback:security:acl principalId $everyone +0ms
  loopback:security:acl accessType * +0ms
  loopback:security:acl permission DENY +0ms
  loopback:security:acl with score: +0ms 7495
  loopback:security:acl ---ACL--- +1ms
  loopback:security:acl model MyModel +0ms
  loopback:security:acl property * +0ms
  loopback:security:acl principalType ROLE +0ms
  loopback:security:acl principalId $everyone +0ms
  loopback:security:acl accessType * +0ms
  loopback:security:acl permission DENY +0ms
  loopback:security:acl with score: +0ms 7495
  loopback:security:acl ---Resolved--- +0ms
```

Why is it that the relevant ACL is not being searched when I use GET?

**Comment by loay** [comment]
Can you pleas provide more info or a repo to reproduce the issue so I can test on my end?

**Comment by loay** [comment]
Hello,
I am closing this issue for now. 
If you are still running into problems, feel free to leave a comment and I will reopen the issue.
closes#1686
